advisor/src/main/kotlin/advisors/NexusIq.kt

```diff
@@ -51,13 +52,16 @@ private const val REQUEST_CHUNK_SIZE = 100
  */
 class NexusIq(
     name: String,
-    config: AdvisorConfiguration
+    config: AdvisorConfiguration,
```

Comment:
A minor improvement IMHO would be if the class was passed the concrete `NexusIqConfiguration` as argument rather than the whole `AdvisorConfiguration`. The factory would do the check whether the `nexusIq` property in `AdvisorConfiguration` is defined and pass it to the newly created instance. The `nexusIqConfig` property in the class would then no longer be necessary.

Reply:
I kind of agree with you, however, that does not get along well with the advisors being chrildren of the `Advisor` class. This class requires an `AdvisorConfiguration` field which is also used in one its methods. Please let me know if you have a good solution to overcome this, since I am not totally satisfied with my solution.

Reply:
BTW, please drop the trailing comma.

Reply:
Also, it just occurred to make that (also before this change) we have the general limitation that not multiple advisors could be enabled / configured. That is also a limitation that we should lift. Not necessarily as part of this PR, though.

Reply:
> 
> 
> I kind of agree with you, however, that does not get along well with the advisors being chrildren of the `Advisor` class. This class requires an `AdvisorConfiguration` field which is also used in one its methods. Please let me know if you have a good solution to overcome this, since I am not totally satisfied with my solution.

I see, the advisor configuration is part of the result produced by the advisor. With this new approach of handling configuration, this is a bit cumbersome: the configuration stored in the result would contain not only the configuration of this concrete advisor instance but the whole advisor configuration; and if we add support for multiple advisors (as mentioned by @sschuberth), this configuration would be duplicated for each advisor that is running.

One solution could be that the concrete advisor implementations are passed their specific configuration, but to the base class they pass a newly created `AdvisorConfiguration` that only contains their configuration. This is not super elegant either, but at least the relevant configuration can be stored in the result without duplication and without the need for Jackson annotations.

Reply:
I agree we need to find a solution for
1) not duplicating the whole advisor configuration per advisor in the result file,
2) the ugly `null`-check in line 62 below (by only creating advisors for which we *do* have a config in the first place).

Reply:
We decided to merge this in its current state and address the above-mentioned issues when working on https://github.com/oss-review-toolkit/ort/issues/3638.